src/app/screens/confirmBrc20Transaction/index.tsx

```diff
@@ -0,0 +1,138 @@
+import BottomBar from '@components/tabBar';
+import styled from 'styled-components';
+import useDebounce from '@hooks/useDebounce';
+import useWalletSelector from '@hooks/useWalletSelector';
+import { BRC20ErrorCode } from '@secretkeylabs/xverse-core/transactions/brc20';
+import { brc20TransferEstimateFees, CoreError } from '@secretkeylabs/xverse-core';
+import {
+  getFeeValuesForBrc20OneStepTransfer,
+  ConfirmBrc20TransferState,
+  ExecuteBrc20TransferState,
+} from '@utils/brc20';
+import { isInOptions, isLedgerAccount } from '@utils/helper';
+import { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useResetUserFlow } from '@hooks/useResetUserFlow';
+import { useTranslation } from 'react-i18next';
+import ConfirmBrc20TransactionComponent from './confirmBrc20TransactionComponent';
+import { OnChangeFeeRate } from './editFees';
+
+const ScrollContainer = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  overflow-y: auto;
+  &::-webkit-scrollbar {
+    display: none;
+  }
+  height: 600px;
+  width: 360px;
+  margin: auto;
+`;
+
+export function ConfirmBrc20Transaction() {
+  const { t } = useTranslation('translation', { keyPrefix: 'CONFIRM_BRC_20' });
+  const { selectedAccount } = useWalletSelector();
+  const navigate = useNavigate();
+  const {
+    recipientAddress,
+    estimateFeesParams,
+    estimatedFees: initEstimatedFees,
+    token,
+  }: ConfirmBrc20TransferState = useLocation().state;
+  const { subscribeToResetUserFlow } = useResetUserFlow();
+  useEffect(() => subscribeToResetUserFlow('/confirm-brc20-tx'), [subscribeToResetUserFlow]);
+
+  const [userInputFeeRate, setUserInputFeeRate] = useState('');
+  const [estimatedFees, setEstimatedFees] = useState(initEstimatedFees);
+  const [isFeeLoading, setIsFeeLoading] = useState(false);
+  const [error, setError] = useState<BRC20ErrorCode | ''>('');
+  const debouncedUserInputFeeRate = useDebounce(userInputFeeRate, 300);
+
+  const { txFee, inscriptionFee, totalFee, btcFee } = getFeeValuesForBrc20OneStepTransfer(
+    estimatedFees.valueBreakdown,
+  );
+
+  useEffect(() => {
+    const runEstimate = async () => {
+      setIsFeeLoading(true);
+      setError('');
+      try {
+        const result = await brc20TransferEstimateFees({
+          ...estimateFeesParams,
+          feeRate: Number(debouncedUserInputFeeRate),
+        });
+        setEstimatedFees(result);
+      } catch (err) {
+        const e = err as Error;
+        if (CoreError.isCoreError(e) && (e.code ?? '') in BRC20ErrorCode) {
+          setError(e.code as BRC20ErrorCode);
+          // TODO add logic in from core hook to show what the fee would have been
+        } else {
+          setError(BRC20ErrorCode.SERVER_ERROR);
+        }
+      }
+      setIsFeeLoading(false);
+    };
+    if (debouncedUserInputFeeRate) {
+      runEstimate();
+    }
+  }, [debouncedUserInputFeeRate, estimateFeesParams]);
```

Comment:
🤔 why not use the fee estimate hook from Core?

Reply:
I wanted to use the hook from core (including the error handling), but because of the design, I need to run the estimate fee:

-  once in the send-brc20 screen to validate whether we can move to the confirm-brc20 screen
- and not on the confirm-brc20 screen on first render (because we have no user input yet), but again on change of the user input field

I couldn't control when the core hook runs because of the 'no conditional hooks' rule, hence line 79-81

Reply:
Ah, ok, cool. Maybe we can refactor that hook somehow 🤔

Reply:
After sleeping on this, I actually think it might be better to use the hook here 🙈 

Firstly, it would clean up this chunk of code into a single line of code at the cost of 1 fee calc which we would do on a fee rate change anyways (not super expensive).

Secondly, it would compartmentalise this component to calculate its own state instead of relying on an external input to tell it what the initial fee would be, removing the potential for a bug coming in from an external component.

What do you think?

Reply:
IMO the best way would be to expose the estimateFee function from the hook instead of calling it inside a useEffect, in the same way that the execute hook is structured. then we get the code encapsulation on both send and confirm screens, as well as the option to refetch on first render of confirm.

I don't have a huge problem with refetching on first render of confirm screen, except that this pattern of passing state around with location.state is the established pattern throughout the app. IMO that shouldn't be the pattern, and we could be using react context API for flows which couple screens together like this, but that's a new pattern that I would want to read more code before I tackle.

Reply:
Yeah, that's a really good idea 👍 I think the state stuff makes sense in React Native but not really for web.

For the hook, what do you think about adding a boolean arg called `skipInitialFetch` which defaults to false? Then you can set it to true here and get the desired outcome.

Reply:
sounds good 👍

Reply:
Cool, will add it now

Reply:
Ok, it's in the latest version 👍